File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int a = 0;
        int b = 0;
        int c = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 0) {
                a++;
            } else if (nums[i] == 1) {
                b++;
            } else {
                c++;
            }
        }
        int i = 0;
        while (a != 0) {
            nums[i] = 0;
            a--;
            i++;
        }
        while (b != 0) {
            nums[i] = 1;
            b--;
            i++;
        }
        while (c != 0) {
            nums[i] = 2;
            c--;
            i++;
        }
    }
};

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: list[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        idx=0
        for i in range(len(nums)):
            if nums[i]==0:
                nums[idx],nums[i]=nums[i],nums[idx]
                idx+=1

        for i in range(len(nums)):
            if nums[i]==1:
               nums[idx],nums[i]=nums[i],nums[idx]
               idx+=1
               
        